packages/react-router/src/useRenderedNavigationItems.tsx

```diff
@@ -66,8 +66,18 @@ export function useRenderedNavigationItems(
     renderSection: RenderSectionFunction
 ) {
     return useMemo(() => {
+        const clonedItems = [...navigationItems];
+
         // Highest priority is rendered first.
-        const sortedItems = [...navigationItems]
+        const sortedItems = clonedItems
+            .map(x => {
```

Comment:
Since your now looping your item in a .map, i don't think cloning the array is necessary, since the result of the .map is a new array reference